vaadin-testbench-integration-tests-junit5/src/test/java/com/vaadin/tests/WaitForVaadinIT.java

```diff
@@ -38,6 +41,18 @@ public void waitForVaadin_activeConnector_waits() {
         assertExecutionBlocked(() -> getCommandExecutor().waitForVaadin());
     }
 
+    @BrowserTest
+    public void waitForVaadin_activeConnector_waitsUtilReady() {
+        openTestURL();
+        assertDevServerIsNotLoaded();
+        getCommandExecutor().executeScript(
+                "window.Vaadin.Flow.clients[\"blocker\"] = {isActive: () => true};");
+        setWaitForVaadinLoopHook(500,
+                "window.Vaadin.Flow.clients[\"blocker\"] = {isActive: () => false};");
```

Comment:
This might be done with executing the JS `"window.Vaadin.Flow.clients[\"blocker\"] = {isActive: () => false};"` as a callback to `setTimeout()` method, so the clients get non-active after some timeout threshold. Then you don't need a reflection and waitForVaadin hook.

Reply:
`waitForVaadin` blocks both server and browser very effiently. `setTimeout` (and also I've tried `setInterval`, `requestAnimationFrame`) are only executed after while loop is done, which is too late.

Reply:
Got it, thanks.